flink-runtime/src/main/java/org/apache/flink/runtime/security/token/KerberosDelegationTokenManager.java

```diff
@@ -173,27 +274,95 @@ void startTGTRenewal() throws IOException {
                                                     LOG.warn("Error while renewing TGT", e);
                                                 }
                                             }),
-                            tgtRenewalPeriod,
+                            0,
                             tgtRenewalPeriod,
                             TimeUnit.MILLISECONDS);
-            LOG.debug("TGT renewal task started and reoccur in {} ms", tgtRenewalPeriod);
+            LOG.info("TGT renewal task started and reoccur in {} ms", tgtRenewalPeriod);
         } else {
-            LOG.debug("TGT renewal task not started");
+            LOG.info("TGT renewal task not started");
         }
     }
 
+    @VisibleForTesting
     void stopTGTRenewal() {
         if (tgtRenewalFuture != null) {
             tgtRenewalFuture.cancel(true);
             tgtRenewalFuture = null;
         }
     }
 
+    @VisibleForTesting
+    void startTokensUpdate() {
+        try {
+            LOG.info("Starting tokens update task");
```

Comment:
Should we check `isLoginPossible` before starting this logic?

Reply:
This is an internal function and not intended to use from the outside.
In `start` function there is a check:
```
        if (!kerberosRenewalPossibleProvider.isRenewalPossible()) {
            LOG.info("Renewal is NOT possible, skipping to start renewal task");
            return;
        }
```
What started to bother me though is that `kerberosRenewalPossibleProvider.isRenewalPossible` contains more or less the same code just like `kerberosLoginProvider.isLoginPossible`.

I think debug messages must be added to `kerberosLoginProvider.isLoginPossible` and `kerberosRenewalPossibleProvider` can be deleted as-is. WDYT?

Reply:
I've made the changes, plz share your opinion if you have better idea.

Reply:
I agree if they contain the same check/logic we should keep only the `kerberosLoginProvider`

Reply:
looks good